lib/chef/resource/homebrew_install.rb

```diff
@@ -0,0 +1,147 @@
+        HOMEBREW_CACHE = "#{USER_HOME}/Library/Caches/Homebrew".freeze
+
+        # Creating the basic directory structure needed for Homebrew
+        directories = ["bin", "etc", "include", "lib", "sbin", "share", "var", "opt",
+                        "share/zsh", "share/zsh/site-functions",
+                        "var/homebrew", "var/homebrew/linked",
+                        "Cellar", "Caskroom", "Homebrew", "Frameworks"
+                      ].freeze
+        directories.each do |dir|
+          directory "/usr/local/#{dir}" do
+            mode "0755"
+            owner new_resource.user
+            group "admin"
+            action :create
+          end
+        end
+
+        user_directories = ["#{USER_HOME}", "#{USER_HOME}/Library",
+                            "#{USER_HOME}/Library/Caches", "#{USER_HOME}/Library/Caches/Homebrew"
+        ]
+        user_directories.each do |dir|
+          directory "#{dir}" do
+            mode "0755"
+            owner new_resource.user
+            group "admin"
+            action :create
+          end
+        end
+
+        if new_resource.tools_url
+          dmg_package new_resource.tools_pkg_name do
+            source new_resource.tools_url
+            type "pkg"
+          end
+        else
+          build_essential "install Command Line Tools for Xcode" do
+            action :upgrade
+          end
+        end
+
+        script "Download and unpack Homebrew" do
+          interpreter "bash"
+          cwd "/usr/local/Homebrew"
+          code <<-CODEBLOCK
+            git init -q
+            curl #{BREW_REPO} -o brew-master.zip
+            unzip brew-master.zip -d /usr/local/Homebrew
+          CODEBLOCK
+          user new_resource.user
+        end
+
+        script "move files to their correct locations" do
+          interpreter "bash"
+          cwd "/usr/local/Homebrew"
+          code <<-CODEBLOCK
+            mv /usr/local/Homebrew/brew-master/* /usr/local/Homebrew/
+            mv /usr/local/Homebrew/brew-master/.* /usr/local/Homebrew/
+            rmdir /usr/local/Homebrew/brew-master/
+          CODEBLOCK
+          user "root"
+        end
+
+        cmd = Mixlib::ShellOut.new("git", "config", "core.autocrlf", "false", :user => new_resource.user, :environment => nil, :cwd => "/usr/local/Homebrew")
+        cmd.run_command
+
+        cmd = Mixlib::ShellOut.new("ln", "-sf", "/usr/local/Homebrew/bin/brew", "/usr/local/bin/brew", :user => new_resource.user, :environment => nil, :cwd => "/usr/local/Homebrew")
+        cmd.run_command
+
+        cmd = Mixlib::ShellOut.new("/usr/local/bin/brew", "update", "--force", :user => new_resource.user, :environment => nil, :cwd => "/usr/local/Homebrew")
+        cmd.run_command
+
+        local_shell = shell_out("echo $SHELL")
+        if local_shell.stdout.match(/zsh/)
+          shell_out("export PATH="/usr/local/bin:$PATH" >> ~/.zshrc")
+        else
+          shell_out("export PATH="/usr/local/bin:$PATH" >> ~/.bash_profile")
```

Comment:
These two strings aren't valid syntax, you're putting double quotes inside double quotes with no escaping.

Reply:
sorted